test/unit/object/test_base_object.py

```diff
@@ -21,6 +23,19 @@ def test_object_and_response(
     return test_objects_and_responses[request.param]
 
 
+@pytest.fixture(params=('same_file', 'same_folder', 'same_user', 'against_none', 'different_ids', 'different_types'))
+def objects_for_comparison(test_file, test_folder, mock_user, request, mock_box_session):
+    cases = {
+        'same_file': (test_file, test_file, True),
```

Comment:
Because of the way you've defined these first three fixtures, your tests are trivially correct. It would be difficult (though obviously not impossible) to accidentally write an incorrect equality such that `foo == foo` isn't `True`.

The true test of your equality would be if `fooFactory() == fooFactory()` is `True`.

So here, I think what we really want is
```python
'same_file': (test_file, File(test_file.object_id), True),
...
```